src/librustc/traits/error_reporting.rs

```diff
@@ -347,6 +347,52 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
         }
     }
 
+    fn describe_generator(&self, body_id: hir::BodyId) -> Option<&'static str> {
+        self.tcx.hir().body(body_id).generator_kind.map(|gen_kind| {
+            match gen_kind {
+                hir::GeneratorKind::Gen => { "a generator" },
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Block) => "an async block",
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Fn) => "an async function",
+                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Closure) => "an async closure",
+            }
```

Comment:
My preferred style here would be:

```
            match gen_kind {
                hir::GeneratorKind::Gen => "a generator",
                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Block) => "an async block",
                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Fn) => "an async function",
                hir::GeneratorKind::Async(hir::AsyncGeneratorKind::Closure) => "an async closure",
            }
```

Reply:
ah missed removing the curlies around the generator case.